streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -286,6 +286,31 @@ trait ZStreamPlatformSpecificConstructors { self: ZStream.type =>
       .managed(is)
       .flatMap(fromInputStream(_, chunkSize))
 
+  /**
+   * Creates a stream from [[java.io.Reader]].
+   */
+  def fromReader(reader: => Reader): ZStream[Blocking, IOException, Char] =
```

Comment:
Ahoy! Let's make this similar to `fromInputStream`. Important points:
- Chunking: there's a `read` variant that reads into an array buffer
- The by-name reader needs to be captured in value (otherwise it gets re-initialized every time)

Reply:
Ah, good points. I was overeager to make this super simple :). Updated.